addons/account/models/account_move.py

```diff
@@ -1485,7 +1485,16 @@ def _compute_amount(self):
             if new_pmt_state == 'paid' and move.move_type in ('in_invoice', 'out_invoice', 'entry'):
                 reverse_type = move.move_type == 'in_invoice' and 'in_refund' or move.move_type == 'out_invoice' and 'out_refund' or 'entry'
                 reverse_moves = self.env['account.move'].search([('reversed_entry_id', '=', move.id), ('state', '=', 'posted'), ('move_type', '=', reverse_type)])
-                caba_moves = self.env['account.move'].search([('tax_cash_basis_move_id', 'in', move.ids + reverse_moves.ids), ('state', '=', 'posted')])
+                if self.env.company.tax_exigibility and self.env.company.tax_cash_basis_journal_id:
+                    domain = [
+                        ('tax_cash_basis_move_id', 'in', move.ids + reverse_moves.ids),
+                        ('state', '=', 'posted'),
+                        ('move_type', '=', 'entry'),
+                        ('journal_id', '=', self.env.company.tax_cash_basis_journal_id.id)
+                    ]
```

Comment:
```suggestion
                    domain = [
                        ('tax_cash_basis_move_id', 'in', move.ids + reverse_moves.ids),
                        ('state', '=', 'posted'),
                        ('move_type', '=', 'entry'),
                    ]
```
I would remove `journal_id` from the domain since it could be set at some point and modified later in the company settings.
The initial journal would then be ignored (unlikely, but possible :slightly_smiling_face: ).

Reply:
Did not though about that use case, I will remove it to play safe :)